hooks/boot/05-data-helpers.rb

```diff
@@ -0,0 +1,126 @@
+module DataResetHookContextExtension
+  def remote_host?(hostname)
+    !['localhost', '127.0.0.1', `hostname`.strip].include?(hostname)
+  end
+  
+  def load_db_config(db)
+    case db
+    when 'foreman'
+      module_name = 'foreman'
+      user_param = 'username'
+      db_param = 'database'
+      param_prefix = 'db_'
+    when 'candlepin'
+      module_name = 'katello'
+      user_param = 'user'
+      db_param = 'name'
+      param_prefix = 'candlepin_db_'
+    when 'pulpcore'
+      module_name = 'foreman_proxy_content'
+      user_param = 'user'
+      db_param = 'db_name'
+      param_prefix = 'pulpcore_postgresql_'
+    else
+      raise "installer module unknown for db: #{db}"
+    end
+  
+    {
+      host: param_value(module_name, "#{param_prefix}host") || 'localhost',
+      port: param_value(module_name, "#{param_prefix}port") || 5432,
+      database: param_value(module_name, "#{param_prefix}#{db_param}") || db,
+      username: param_value(module_name, "#{param_prefix}#{user_param}"),
+      password: param_value(module_name, "#{param_prefix}password"),
+    }
+  end
+  
+  def pg_command_base(config, command, args)
+    port = "-p #{config[:port]}" if config[:port]
+    "PGPASSWORD='#{config[:password]}' #{command} -U #{config[:username]} -h #{config[:host]} #{port} #{args}"
```

Comment:
The downside of using it like this is that you can see the password in `ps aux`. It would be much better to pass an environment variable via native Ruby methods rather than relying on shell.

Reply:
Checking https://github.com/theforeman/foreman-installer/blob/develop/hooks/boot/01-kafo-hook-extensions.rb#L119 and https://github.com/theforeman/kafo/blob/master/lib/kafo/puppet_command.rb#L49 , I think it makes sense to solve this in a separate PR.